src/app/server/Server.cpp

```diff
@@ -470,8 +470,7 @@ static void ChipEventHandler(const DeviceLayer::ChipDeviceEvent * event, intptr_
         advertise();
         break;
 #if CHIP_DEVICE_CONFIG_ENABLE_THREAD
-    case DeviceLayer::DeviceEventType::kThreadStateChange:
-        VerifyOrReturn(event->ThreadStateChange.AddressChanged);
+    case DeviceLayer::DeviceEventType::kSrpClientRequestComplete:
```

Comment:
The point of this (temporary) code was to update the service instance name after the commissioning (during which a nodeID is assigned to the device). As far as I understand, the new code will start operational advertising immediately after starting the SRP client, right? Have you tested your change with the new commissioning flow using ZCL Network Commissioning Cluster commands and confirmed the advertised node ID is correct?

Reply:
It delays the new advertise to after the `otSrpClientSetHostAddresses` request in  `_SetupSrpHost` is completed succesfully. 

I did not test with the using ZCL Network Commissioning Cluster commands I was inspecting the advertissement with Discovery. 
So you are right this might be wrong and I just saw Lukasz comment aout the Duplicated error. I made that change as a workaround to that Duplicated error but seems like the error is expected.

I will try with the ZCL commissioning Cluster and validate if the node id is correct. I can revert this change.

Reply:
Yeah, if you could find a way to get rid of the "Duplicated" error, that would be awesome, but preserving support for the new commissioning flow should be our priority for now, so thanks in advance for verifying if your changes don't break it.

Reply:
I've tested these changes on NXP platform and there is one thing that I want to mention:

- _SetupSrpHost is never called, that's because in GenericThreadStackManagerImpl_OpenThread<ImplClass>::_OnPlatformEvent(const ChipDeviceEvent * event) the hostInfo->mName is always NULL. The hostname is set when the operational advertisment starts but this never happens because this line "case DeviceLayer::DeviceEventType::kThreadStateChange" was removed.

Reply:
Are you sure doru? You can test on your side by changing back this line 
case DeviceLayer::DeviceEventType::`kSrpClientRequestComplete`: back to what it was : DeviceLayer::DeviceEventType:: `kThreadStateChange`: and you will see.

`kThreadStateChange` still exist and triggers multiple calls on this event. 
The goal here is _SetupSrpHost is triggered on `kThreadStateChange`. Once this setup on srp_client is complete and transaction done with the srp_server we receive `OnSrpClientNotification` you can receive either a succes or an error but you should be received that event. if you have an error it wont trigger the ne advertissement because of where I trigger the event `kSrpClientRequestComplete`. Either way you should still receive a `OnSrpClientNotification` if you don't there is another issues here.

I can confirm that i received `OnSrpClientNotification` every time. 

Now the triggering of the new advertissement may not be the way to go`kSrpClientRequestComplete` as it was a way to not get the duplicated error on my side. I didn't validate Damain request yet because i am running the some test for the TE2. 

But like I said if you think this is your issue, uou can set the advertisse to be called on kThreadStateChange and test it on your platform

Reply:
I reverted the change mentionned here to avoid any problem or confusion for the Test Event. I will explore the duplicated error again after the event